Utils/Functions/Filesystem/Files/script.js

```diff
@@ -2,202 +2,205 @@ var fileAnalyser = Bean("fileAnalyser");
+ * Creates a new temporary file with the given file extension
+ *
+ * @param {string} extension file extension
+ * @returns {string} file path of created temporary file
+ */
+function createTempFile(extension) {
+	return fileUtilities.createTmpFile(extension);
 }
 
 /**
- * creates temporary directory
+ * Creates a new temporary temporary directory
+ *
  * @returns {string} file path of created temporary directory
  */
-function createTempDir() { // eslint-disable-line no-unused-vars
-    var tmpFile = Bean("fileUtilities").createTmpDir();
-    return tmpFile.getAbsolutePath();
-}
-
-/**
-* Checksum prototype
-* @constructor
-*/
-function Checksum() {
-    this._method = "SHA";
-    this._checksumCalculator = Bean("checksumCalculator");
+function createTempDir() {
+	return fileUtilities.createTmpDir();
 }
 
 /**
-* sets wizard
-* @param {SetupWizard} wizard setup wizard
-* @returns {Checksum} Checksum object
-*/
-Checksum.prototype.wizard = function (wizard) {
-    this._wizard = wizard;
-    return this;
-}
-
-/**
-* sets checksum algorithm
-* @param {string} algorithm algorithm (e.g. "SHA")
-* @returns {Checksum} Checksum object
-*/
-Checksum.prototype.method = function (algorithm) {
-    this._method = algorithm;
-    return this;
-}
-
-/**
-* sets file for which the checksum shall be computed
-* @param {string} file file for which the checksum shall be computed
-* @returns {Checksum} Checksum object
-*/
-Checksum.prototype.of = function (file) {
-    this._file = file;
-    return this;
-}
-
-/**
-* returns calculated checksum
-* @returns {string} calculated checksum
-*/
-Checksum.prototype.get = function () {
-    if (this._wizard) {
-        var progressBar = this._wizard.progressBar(tr("Checking file consistency..."));
-    }
-
-    return this._checksumCalculator.calculate(this._file, this._method, function (progressEntity) {
-        if (progressBar) {
-            progressBar.accept(progressEntity);
-        }
-    });
+ * Sets the given file permissions
+ *
+ * @param {string} filePath file path
+ * @param {string} permissions file permissions (e.g. "r--r--r--")
+ * @returns {void}
+ */
+function chmod(filePath, permissions) {
+	fileUtilities.chmod(filePath, permissions);
 }
 
 /**
-* sets file permissions
-* @param {string} filePath file path
-* @param {string} permissions file permissions (e.g. "r--r--r--")
-* @returns {void}
-*/
-function chmod(filePath, permissions) { // eslint-disable-line no-unused-vars
-    var permissionsObj = java.nio.file.attribute.PosixFilePermissions.fromString(permissions);
-    var filePathObj = java.nio.file.Paths.get(filePath);
-    java.nio.file.Files.setPosixFilePermissions(filePathObj, permissionsObj);
+ * Checksum
+ */
+class Checksum {
+	constructor() {
+		this.checksumCalculator = Bean("checksumCalculator");
+		this._method = "SHA";
+	}
+
+	/**
+	 * sets wizard
+	 * @param {SetupWizard} wizard setup wizard
+	 * @returns {Checksum} Checksum object
+	 */
+	wizard(wizard) {
```

Comment:
I think we should rename the fluid style API method to the `withX` syntax

Reply:
Concerning wizard, I'm not sure. Even though, sometimes, it is useful to have the same name for getting and setting the object

Reply:
As far as I understand the set parameters are all internal/private parameters right?
If this is the case we don't need a getter method right?

Reply:
I think that there are some case when you may want to pass the wizard from an object to another

Reply:
The issue is, that the methods not real setter methods, because they return `this`.

In my opinion it would be better to use:

```javascript
withWizard(wizard) {
   this.wizard = wizard;
}
```

Afterwards it is automatically possible to access the given wizard via:

```javascript
checksum.wizard
// or from inside the Checksum class
this.wizard
```

I don't think that this will lead to problems.

Reply:
This is not a real issue IMHO, it is a good pattern to allow chaining stuff. 

I don't really like accessing property from the outside

Reply:
Even in JavaScript?

Reply:
why not?

Reply:
I understand I thought you meant accessor methods, but in this case we're talking directly about accessing the property. Normally I would agree that allowing direct access to the property is bad, but I'm not really a friend of providing an imaginary name for a "private" property and then allowing access to it via a getter accessor method. If you think this is a cleaner solution we can also go with that.

I'm not suggesting to perform these changes in this PR, because it is simply out of scope